lib/final_destination/ssrf_detector.rb

```diff
@@ -61,7 +61,12 @@ def self.ip_allowed?(ip)
     end
 
     def self.lookup_and_filter_ips(name, timeout: nil)
-      ips = lookup_ips(name, timeout: timeout)
+      begin
+        ips = lookup_ips(name, timeout: timeout)
+      rescue SocketError
+        raise LookupFailedError, "FinalDestination: lookup failed"
```

Comment:
Do we want to be more verbose about the failure and include the `name` in the `LookupFailedError` exception message? 🤔 

```ruby
raise LookupFailedError, "FinalDestination: lookup failed for '#{name}'"
```

Reply:
That's a good idea. It didn't cross my mind since we are rescuing it in the flow addressed in this PR, but for any other call site it's probably useful. 👍